tests/tensorflow2/test_keras.py

```diff
@@ -588,6 +588,33 @@ def test_layer_names(out_dir, tf_eager_mode):
         assert re.match(pattern=pattern, string=tname) is not None
 
 
+@pytest.mark.slow
+def test_regex_filtering_for_default_collections(out_dir):
+    hook = smd.KerasHook(
+        out_dir,
+        save_config=SaveConfig(save_interval=9),
+        include_collections=[CollectionKeys.LAYERS, CollectionKeys.GRADIENTS],
+    )
+    hook.get_collection("layers").include("^dense")
+    hook.get_collection("gradients").include("gradients/dense")
+    # hook.get_collection(CollectionKeys.LAYERS).include("dense")
+    helper_keras_fit(
+        out_dir,
+        hook=hook,
+        save_config=SaveConfig(save_interval=10),
+        steps=["train"],
+        run_eagerly=True,
+    )
+
+    tr = create_trial_fast_refresh(out_dir)
+    layer_tnames = tr.tensor_names(collection=CollectionKeys.LAYERS)
+    gradient_tnames = tr.tensor_names(collection=CollectionKeys.GRADIENTS)
+    assert len(layer_tnames) == (8 if is_tf_2_2() else 0)
+    assert len(gradient_tnames) == (4 if is_tf_2_2() else 0)
```

Comment:
Can we make this check stronger and verify that the captured tensor names actually match the regex that was specified for each of the collections?

Reply:
See the lines below this.
We cannot have hardcoded string values simply because the names of the tensors are assigned numerical prefixes based on when the test is executed.